### ISSUE TITLE
Provide a way to force url hash implementation

### ISSUE THREAD

**Issue by albrow** [go-humble/router#4]
Currently, if the browser you're using supports pushState, router uses it. Always. This should be configurable.

I can imagine a use case where pushState is supported but you would still want to use the hash url method. For example, if you're writing a single page application, the hash url method is easier to implement, because your site can just be a single static file. If you want to use pushState for a SPA, you would need a server sitting in front of your static files which always serves an index page.

In other words, it's easier to implement a server that responds correctly to `/#/users/{userId}` than one that responds to `/users/{userId}` for any arbitrary userId. Furthermore, the hash method works when you want to serve your static assets from a cdn, whereas the pushState method typically would not.

**Comment by albrow** [comment]
Fixed in fce4304e6035cf84fad4513ee2bade650233dee3
